compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -1007,8 +1008,19 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
     }
 
     /// Iterates over all the stability attributes in the given crate.
-    fn get_lib_features(self, tcx: TyCtxt<'tcx>) -> &'tcx [(Symbol, Option<Symbol>)] {
-        tcx.arena.alloc_from_iter(self.root.lib_features.decode(self))
+    fn get_lib_features(self, _tcx: TyCtxt<'tcx>) -> LibFeatures {
+        let mut features = LibFeatures::default();
+        for (symbol, stability) in self.root.lib_features.decode(self) {
+            match stability {
+                FeatureStability::AcceptedSince(since) => {
+                    features.stable.insert(symbol, (since, DUMMY_SP));
+                }
+                FeatureStability::Unstable => {
+                    features.unstable.insert(symbol, DUMMY_SP);
+                }
```

Comment:
Maybe it could directly be serialized into the crate metadata in this format?

Reply:
Seems less beneficial to store a bunch of DUMMY_SP in the metadata. Code is simpler now that cjgillot's comment above is applied.